### PR TITLE
Move Python script over to Dart.

### PR DESCRIPTION
**Pull request by TheBosZ**
Per [conversation here](https://groups.google.com/forum/#!topic/flutter-dev/EpYu_IvYl5s).

I don't know if it's viable to make this a command for the Flutter tool as the tool needs `pub get` before it can be run creating a chicken and egg problem.

### CODE REVIEW

File: dev/update_packages.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+final String binaryName = Platform.isWindows ? 'pub.bat' : 'pub';
+update(Directory directory) {
+    for(FileSystemEntity dir in directory.listSync() ) {
```
Comment (Hixie) [inline dev/update_packages.dart]: space after `for`

Comment (Hixie) [inline dev/update_packages.dart]: nit: no space after `listSync()`

Comment (Hixie) [inline dev/update_packages.dart]: nit: indent 2, not 4

File: dev/update_packages.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        if (dir is Directory) {
+            print("Updating ${dir.path}...");
+            Process.runSync(binaryName, ['get'], workingDirectory: dir.path);
+        }
+    }
+}
+
+main() {
+    var FLUTTER_ROOT = new File(Platform.script.toFilePath()).parent.parent.path;
```
Comment (Hixie) [inline dev/update_packages.dart]: nit: s/var/File/

**Comment by Hixie** [comment]
LGTM with the minor style nits.

Thanks!!

**Comment by Hixie** [comment]
(if you don't have permission to merge, just upload the updated code and I'll merge it for you. Thanks again!)

File: dev/update_packages.dart (hunk 2 above)
Comment (TheBosZ) [inline dev/update_packages.dart]: That's actually a String.

Comment (Hixie) [inline dev/update_packages.dart]: All the more reason to be explicit. :-D

**Comment by TheBosZ** [comment]
Updated to fix the nits

**Comment by Hixie** [comment]
Thanks!!

**Comment by TheBosZ** [comment]
:+1:

**Pull request merged**